### DISCUSSION TITLE
How to use the known position of peaks for peak fitting?

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#67] [Q&A] [answered]
I saw that I can use .find_peak_multipeak() on my spectrum to find some peak candidates and then use the result as candidates when I use .fit_peak_multimodel(). 
However, if I know the (aproximate) position of the peaks that should be in the spectrum, how could I use this knowledge to do peak fitting with .fit_peak_multimodel()? Is there a way?

Thank you in advance for your time,
nico

**Comment by georgievgeorgi** [comment]
`.fit_peak_multimodel()` require a parameter `candidates` of type `ListPeakCandidateMultiModel` in order to build the model and to calculate the initial guess. An instance of `ListPeakCandidateMultiModel` can be created manually, but it is much more practical to use `.find_peak_multipeak()` create it automatically.
There is this example: `examples/fit_spe_composite_model.ipynb` that describes how it works.
If there is a problem please let me know.

**Reply by nicocopez** [reply]
Thank you for the quick answer Georgi!
I already tried to use `.find_peak_multipeak()` (actually following your example), but I am unable to properly catch all the peaks that I need. I will keep on playing with the paremeters. In any case, if it is not too complicated to explain, could you tell me how can I create manually my own `ListPeakCandidateMultiModel` ? Alternatively, it would also be very useful if one can choose the range in which the peak should be found.

**Comment by georgievgeorgi** [comment, marked as answer]
This example plots manually created candidates.
```python
from ramanchada2.misc.types import ListPeakCandidateMultiModel
cand = ListPeakCandidateMultiModel.validate([
    {'base_slope': -.1, 'base_intersept': 100, 'boundaries': [0, 300], 'peaks': [
        {'amplitude': 100, 'position': 50, 'sigma': 30, 'skew': 0},
        {'amplitude': 10, 'position': 150, 'sigma': 30, 'skew': .5},
        {'amplitude': 200, 'position': 250, 'sigma': 30, 'skew': -.5},
    ], },
    {'base_slope': .1, 'base_intersept': -50, 'boundaries': [500, 800], 'peaks': [
        {'amplitude': 100, 'position':550, 'sigma': 30, 'skew': 0},
        {'amplitude': 10, 'position': 650, 'sigma': 30, 'skew': .5},
        {'amplitude': 200, 'position': 700, 'sigma': 30, 'skew': -.5},
    ], },
])
cand.plot()
```

If you are interested in the candidates only in a specified range, you could do something like:
```python
cand = spe.trim_axes('x-axis', (580, 600)).find_peak_multipeak()
spe.fit_peak_multimodel(candidates=cand)
```

**Reply by georgievgeorgi** [reply]
If needed we can have sum operation in context of candidates. For example:
```python
cand_auto = spe.find_peak_multipeak(...)
cand_manual = ListPekCandidateMultiModel.validate(...)
cand_all = cand_auto + cand_manual
fit_res = spe.fit_peak_multimodel(candidates=cand_all, ...)
```
